### DISCUSSION TITLE
Low FPS using AMF as host at desktop

### DISCUSSION THREAD

**Discussion by zycboss** [LizardByte/community#200] [Sunshine]
I am using the newest v0.18.4 version and I still see FPS drop on desktop and high latency from an AMF host, statistic shows the FPS can drop to below 10 when there is no movement on screen.

On the same client, when I'm streaming from another host machine using Nvidia GPU and Shield, its desktop FPS is constant 60 and the latency is significantly lower than Sunshine.

Is there any option or settings that I need to change in the config page to get the same constant 60 FPS in Sunshine?
Since I looked through issues and found this problem should have been fixed...

Fix?: [#18](https://github.com/LizardByte/build-deps/pull/18#) 

Related issue: #122 #387 #412 #485

**Comment by ReenigneArcher** [comment]
GFE doesn't work with AMD cards, so that's not worth comparing.

You can try adjusting the various encoder settings in the Web UI. We try to set defaults that give best performance, but these aren't always the best settings for everyone.

**Reply by zycboss** [reply]
> GFE doesn't work with AMD cards, so that's not worth comparing.
> 
> You can try adjusting the various encoder settings in the Web UI. We try to set defaults that give best performance, but these aren't always the best settings for everyone.

Yes, I can understand AMD card works differently from Nvidia, but I tried different combinations in AMF settings and still cannot get a constant FPS like GFE dose by default.

I am attaching my config file here.
```
nv_preset = p4
upnp = enabled
amd_coder = cabac
nv_coder = auto
vt_software = auto
amd_quality = balanced
origin_web_ui_allowed = lan
min_log_level = 2
dwmflush = enabled
nv_rc = cbr
hevc_mode = 2
encoder = amdvce
nv_tune = ull
vt_coder = auto
sunshine_name = WORK
gamepad = x360
qsv_coder = auto
amd_vbaq = disabled
origin_pin_allowed = lan
amd_usage = ultralowlatency
resolutions = [
    1920x1080,
    3440x1440,
    1920x1200,
    3860x2160,
    2560x1440,
    2560x1600
]
key_rightalt_to_key_win = disabled
vt_realtime = enabled
qsv_preset = medium
fps = [30,60,90,120]
amd_preanalysis = disabled
amd_rc = cqp
```

**Reply by zycboss** [reply]
> I wouldnt bother. People have told them multiple times that there's something wrong with sunshine on AMD cards on sunshine. But the devs dont want to hear it, they act like its working perfectly without actually testing. Sunshine will use 50% of an amd video card just on desktop and lowers your fps about 30-40 fps. No way around it until they optimize and figure out why.
> 
> They love to bring up "oh dont compare with nvidia" or " oh dont compare with parsec" instead of admitting their software might actually be at fault

I don't want to be a jerk who complains about something that I am using for free, while the devs did contribute a lot of their work to it.
However, through a few conversations with the devs these days, I found they really don't listen to users feedback.

I can understand that the software is still in 0.x version and not fully completed, but does this mean that the most important thing to do now is to make sure the basic streaming function is working properly? 

Instead, the new versions are just keeping bring new features with no actual improves, and when users giving advices to the new features (which changed user habit or simply unpolished), they will say "that is not a problem to me" or "it is intended to be like this". Their attitude of not listen is a big problem.

If the dev team of Sunshine keep working like this, I am very doubting it can be a good replacement for Parsec, and Moonlight will also come to an end when Nvidia finally removed streaming function from the PC driver.

I'd wish devs from Moonlight can come and take over some works of Sunshine to help them out. It's still a good project though.

**Comment by ReenigneArcher** [comment]
@zycboss I'm guessing you are somewhat new to Sunshine? Those issues you referenced were greatly improved in v0.18.0. There was high delay when streaming mostly static desktop when using any graphics card.

I personally can use AMF from integrated ryzen 5 cpu and I don't experience your issue. This is not me saying the issue is not valid, I'm just saying it's not occurring in a scenario that I can test. This could be an issue that relates to your specific GPU or series of GPU?


> through a few conversations with the devs these days

Was this conversation with me or who specifically? I would like to know so we can improve. If you can point me to a specific example that would also be helpful.


> I'd wish devs from Moonlight can come and take over some works of Sunshine to help them out

The dev of Moonlight does help here when he can, but try to understand there are a lot of users of these applications and it's difficult to keep up with everyone's questions and complaints. We all do this for free in our "spare" time. And since the Nvidia announcement in December, activity from users has increased exponentially; however only a couple of new people helping with development.

Sunshine is not a perfect software for sure. I apologize that we are bringing new features without fixing enough issues. Of course if we knew how to fix every "issue", they would all be fixed.

**Reply by ReenigneArcher** [reply]
P.S. I've opened a 3rd issue for this topic #1143 with a reference to a discussion in the AMF repo. There's probably enough information now to hopefully 🤞 solve these issues, but understand that my role in Sunshine is mostly devops so another dev will need to take an interest in this issue. I think we have 2 regular contributors who are capable of tackling this issue, and they're both aware of it.

[a comment by an automated account is not shown]
